src/foam/nanos/auth/Group.js

```diff
@@ -45,7 +45,12 @@ foam.CLASS({
       class: 'FObjectArray',
       of: 'foam.nanos.auth.Permission',
       name: 'permissions',
-      documentation: 'Permissions set on group.'
+      documentation: 'Permissions set on group.',
+      preSet: function(oldVal, newVal) {
+        // if old value is undefined, that means save action triggered
+        return oldVal !== undefined ? newVal :
```

Comment:
What is this check for? Why not just always filter the newVal?

Reply:
When we click 'Add Item' button, it will tigger FobjectArrayView 'addItem' action. It will clone the object, and call preSet method eventually.
So if we always filter the newVal, you can never add a new item after you click 'Add Item' button.